src/template_parser.rs

```diff
@@ -160,28 +160,46 @@ impl error::Error for TemplateParseError {
     }
 }
 
-fn parse_string_literal<'a, C>(
+fn parse_string_literal<'a, C: 'a>(
     pair: Pair<Rule>,
-    _parse_keyword: &impl Fn(Pair<Rule>) -> TemplateParseResult<PropertyAndLabels<'a, C>>,
+    parse_keyword: &impl Fn(Pair<Rule>) -> TemplateParseResult<PropertyAndLabels<'a, C>>,
 ) -> TemplateParseResult<Expression<'a, C>> {
+    fn flush_text<C>(expressions: &mut Vec<Expression<'_, C>>, text: &mut String) {
+        let term = PropertyAndLabels(Property::String(Box::new(Literal(mem::take(text)))), vec![]);
+        expressions.push(Expression::Property(term));
+    }
+
     assert_eq!(pair.as_rule(), Rule::literal);
+    let mut expressions = Vec::new();
     let mut text = String::new();
     for part in pair.into_inner() {
         match part.as_rule() {
+            Rule::template => {
+                flush_text(&mut expressions, &mut text);
+                expressions.push(parse_template_rule(part, parse_keyword)?);
+            }
             Rule::raw_literal => {
                 text.push_str(part.as_str());
             }
             Rule::escape => match part.as_str().as_bytes()[1] as char {
                 '"' => text.push('"'),
                 '\\' => text.push('\\'),
                 'n' => text.push('\n'),
+                '{' => text.push('{'),
```

Comment:
is a rule for `}` needed?

I think you should add a test for something like

$ jj log -T '"{change_id.short()} \{{description.first_line()}\}\n"'